app/src/main/java/com/example/garden_planner/adapters/ReminderAdapter.java

```diff
@@ -129,7 +140,59 @@ public void bind(Reminder reminder){
 
             tvReminderText.setText(reminder.getReminderMessage());
 
+            completeReminderButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    // create a harvest reminder if the reminder completed is a planting reminder
+                    if (reminder.getReminderType().equals("plant")) {
+                        PlantInBed plantInBed = reminder.getRemindWhichPlant();
+                        plantInBed.setPlantDate(new Date());
+                        plantInBed.setShouldHarvestDate(GardenMethodHelper.convertToDate(
+                                GardenMethodHelper.convertToLocalDateViaInstant(
+                                                plantInBed.getPlantDate())
+                                        .plusWeeks(plantInBed.getPlantType().getHarvestTime())));
+
+                        Reminder harvestReminder = new Reminder();
+                        Date startHarvestWeek = plantInBed.getShouldHarvestDate();
+                        Date endHarvestWeek = GardenMethodHelper.convertToDate(
+                                GardenMethodHelper.convertToLocalDateViaInstant(
+                                        startHarvestWeek).plusWeeks(1));
+
+                        harvestReminder.setReminderStart(startHarvestWeek);
+                        harvestReminder.setReminderEnd(endHarvestWeek);
+                        harvestReminder.setReminderTitle("Harvest " + plantInBed.getDisplayName());
+                        harvestReminder.setReminderMessage(plantInBed.getPlantType().getHarvestAdvice());
+                        harvestReminder.setRemindWhat(plantInBed.getGarden());
+                        harvestReminder.setRemindWhichPlant(plantInBed);
+                        harvestReminder.setRemindWho(ParseUser.getCurrentUser());
+                        harvestReminder.setReminderType("harvest");
```

Comment:
Do you need to set these everytime you make a reminder? Maybe another helper for this if you do this on several places?

Reply:
that's a good idea - I'll set that up